File: tarea7/ruby/Recorrido.rb
require_relative 'Nodo'

def preorden(a)
  if(a != nil)
    print "#{a.getValorNodo()} "
    preorden a.getSubArbolIzq()
    preorden a.getSubArbolDer()
  end
end

def inorden(a)
  if(a != nil)
    inorden(a.getSubArbolIzq())
    print "#{a.getValorNodo()} "
    inorden(a.getSubArbolDer())
  end
end

def postorden(a)
  if(a != nil)
    postorden(a.getSubArbolIzq());
    postorden(a.getSubArbolDer());
    print "#{a.getValorNodo()} "
  end
end

File: tarea7/java/arbol/recorrido.java
package arbol;

class recorrido{
  public static void preorden(Nodo a){
    if(a != null){
      System.out.print(a.getValorNodo()+ " ");
      preorden(a.getSubArbolIzq());
      preorden(a.getSubArbolDer());
    }
  }

  public static void inorden(Nodo a){
    if(a != null){
      inorden(a.getSubArbolIzq());
      System.out.print(a.getValorNodo()+ " ");
      inorden(a.getSubArbolDer());
    }
  }

  public static void postorden(Nodo a){
    if(a != null){
      postorden(a.getSubArbolIzq());
      postorden(a.getSubArbolDer());
      System.out.print(a.getValorNodo()+ " ");
    }
  }
}
